cmd/xl-v1-object.go

```diff
@@ -461,13 +462,16 @@ func (xl xlObjects) PutObject(bucket string, object string, size int64, data io.
 	}
 
 	uniqueID := mustGetUUID()
-	tempErasureObj := path.Join(uniqueID, "part.1")
 	tempObj := uniqueID
 
 	// Initialize md5 writer.
 	md5Writer := md5.New()
 
-	writers := []io.Writer{md5Writer}
+	// Initialize md5 writer for parts, it will be
+	// reset before reading each new part
+	partMD5Writer := md5.New()
```

Comment:
i think leaving out part's md5sum will be better as it does not serve any purpose and it would just consume extra cpu.